Sources/StreamChatUI/Utils/UIViewController+Extensions.swift

```diff
@@ -19,4 +19,76 @@ extension UIViewController {
         view.removeFromSuperview()
         removeFromParent()
     }
+
+    var isUIHostingController: Bool {
+        String(describing: self).contains("UIHostingController")
+    }
+
+    /// Helper method to correctly setup navigation of parent.
+    /// Used when view is wrapped using `UIHostingController` and used in SwiftUI
+    /// so all properties of `navigationItem` are propagated correctly.
+    @available(iOS 13.0, *)
+    func setupParentNavigation(parent: UIViewController) {
+        let parentNavItem = parent.navigationItem
+
+        if parentNavItem.backBarButtonItem == nil {
+            parentNavItem.backBarButtonItem = navigationItem.backBarButtonItem
+        }
+
+        if #available(iOS 14.0, *) {
+            parent.navigationItem.backButtonDisplayMode = navigationItem.backButtonDisplayMode
+        }
+
+        if parentNavItem.backButtonTitle == nil {
+            parentNavItem.backButtonTitle = navigationItem.backButtonTitle
+        }
+
+        if parentNavItem.compactAppearance == nil {
+            parentNavItem.compactAppearance = navigationItem.compactAppearance
+        }
+
+        parent.navigationItem.hidesSearchBarWhenScrolling = navigationItem.hidesSearchBarWhenScrolling
+
+        if parent.navigationItem.largeTitleDisplayMode == .automatic {
+            parentNavItem.largeTitleDisplayMode = navigationItem.largeTitleDisplayMode
+        }
+
+        if parent.navigationItem.hidesBackButton == false {
+            parent.navigationItem.hidesBackButton = navigationItem.hidesBackButton
+        }
+
+        if let leftBarButtonItems = parentNavItem.leftBarButtonItems, leftBarButtonItems.isEmpty {
+            parentNavItem.leftBarButtonItems = navigationItem.leftBarButtonItems
+        }
+
+        parent.navigationItem.leftItemsSupplementBackButton = false
```

Comment:
Can you add a comment explaining why this is hardcoded?

Reply:
It obviously shouldn't be, fixed.